tests/test.hs

```diff
@@ -432,17 +435,37 @@ selfTests
 
 testGroupsWithLibs :: String -> DependentTests -> [TestTree]
 testGroupsWithLibs name (DependentTests libTests nonlibTests) =
-  [ testGroup (name <> "Lib") libTests
-  , after AllFinish (name <> "Lib") $ testGroup name nonlibTests
+  [ testGroup (name <> "-libs") libTests
+  , after AllFinish (name <> "-libs") $ testGroup name nonlibTests
   ]
 
+-- | Creates a [TestTree] that runs without parallelism
+testSequentially :: String -> SequentialTests -> [TestTree]
+testSequentially name (SequentialTests indexedTests) =
+  let grouped = (\(t, n) -> (testGroup (name <> show n) [t], n)) <$> indexedTests
+      -- turns i.e. [a, b, c, d] into [(a, b), (b, c), (c, d)]
+      pairs = zip <*> tail $ grouped
+      deps = (\((_, n1), (t2, _)) -> after AllFinish (name <> show n1) t2) <$> pairs
+  in
+    case grouped of
+      [] -> []
+      (t:_) -> fst t : deps
```

Comment:
Oddly, no.  I tried this, and it ran on all my CPUs regardless.  However, doing `stack --nix test -ta '-j1 -p sequential'` DID work.  I do like the idea of adding `sequential` to the test name, however, so that you can run the sequential tests by name.